### DISCUSSION TITLE
How to get the histogram of coordination number？

### DISCUSSION THREAD

**Discussion by Devin-one** [PMEAL/OpenPNM#2670] [Q&A]
Hello, professor. What codes can I use to get thes histogram of coordination number? Is it necessary to add coordination number in the geometry model？Thank you in advance for your reply.

**Comment by jgostick** [comment]
You can find the number of neighbors to a given pore using `pn.num_neighbors(pores=1)`, and you can also get the result for ALL pores as `z = pn.num_neighbors(pores=pn.Ps)`.  The default format for the returned result is a list of the number of pores neighboring each of the given inputs.  So you can then just to `plt.hist(z)`.

**Reply by jgostick** [reply]
I should add that in V3 we started including a model on the network for `pore.coordination_number` which is *not* run, but you can run it using `pn.regenerate_models()`, then you'll have `pn['pore.coordination_number']` on your network which you can then use as `plt.hist(pn['pore.coordination_number'])`.

**Comment by budidharmala** [comment]
Hallo profesor, 
I have input in the form of an image sequence from a rock sample. How can I calculate the 3D coordination number using OpenPNM? Also, how can I visualize it?

**Reply by jgostick** [reply]
I just wrote a long-overdue FAQ on network extraction in PoreSpy and analysis in OpenPNM.  Please give [this](https://github.com/PMEAL/OpenPNM/discussions/2867) a read and see if that gets you started.

**Reply by budidharmala** [reply]
oke, thank you professor, i will start it.

**Reply by budidharmala** [reply]
Hello Professor,

First of all, I would like to thank you very much for your help.

I have succeeded in creating binary images in .tiff 

in this form [here](https://openpnm.org/examples/applications/network_extraction.html) i see the format of images in .npz.

`path = Path(os.getcwd(),
            '../../tests/fixtures/berea_100_to_300.npz')
data = np.load(path.resolve())
im = data['im']`

Should I convert from .tiff to .npz?

and to calculate 3D coordination numbers, which format should I use?

Regards,
Bd

**Reply by jgostick** [reply]
npz is just a file format for storing a numpy array in a compressed form, which is preferred for our online examples.  Whether using tiff -> numpy array or npz -> numpy array doesn't matter.

BTW, these questions are starting to fall under the category of "you need to learn more about Python before you ask the devs of the packages you're using for help".  Try google or chatgpt first.  I don't mind answering questions about openpnm and porespy, but my time is limited.
